tests/test_ioc_system.py

```diff
@@ -324,71 +265,87 @@ async def test_create_softioc_time_epics_changes(
-
-    # Trigger a _RecordUpdater.update(), to do a Put command
-
-    dummy_server_system.expected_message_responses.update(
-        {"PCAP1.TRIG_EDGE=Either": "OK"}
-    )
-
-    await caput(TEST_PREFIX + ":PCAP1:TRIG_EDGE", "Either", wait=True, timeout=TIMEOUT)
-
-    # Confirm the server received the expected string
-    assert not dummy_server_system.expected_message_responses
 
-    dummy_server_system.send += ["!PCAP1.TRIG_EDGE=Either\n.", ".", "."]
+    try:
+        arm_queue = asyncio.Queue()
+        m1 = camonitor(TEST_PREFIX + ":PCAP:ARM", arm_queue.put, datatype=str)
+        assert await asyncio.wait_for(arm_queue.get(), TIMEOUT) == "0"
 
-    async def expected_messages_received():
-        """Wait until the expected messages have all been received by the server"""
-        while len(dummy_server_system.send) > 2:
-            await asyncio.sleep(0.1)
+        await caput(TEST_PREFIX + ":PCAP:ARM", 1, wait=True, timeout=TIMEOUT)
 
-    await asyncio.wait_for(expected_messages_received(), timeout=TIMEOUT)
+        assert await asyncio.wait_for(arm_queue.get(), TIMEOUT) == "1"
+    finally:
+        m1.close()
 
 
 @pytest.mark.asyncio
-async def test_bobfiles_created(dummy_server_system: DummyServer, subprocess_ioc: Path):
-    bobfile_temp_dir = subprocess_ioc
+async def test_bobfiles_created(mocked_panda_standard_responses):
+    bobfile_temp_dir, *_ = mocked_panda_standard_responses
     assert bobfile_temp_dir.exists() and BOBFILE_DIR.exists()
     old_files = os.listdir(BOBFILE_DIR)
     for file in old_files:
-        assert filecmp.cmp(
-            f"{bobfile_temp_dir}/{file}", f"{BOBFILE_DIR}/{file}"
-        ), f"File {bobfile_temp_dir/file} does not match {BOBFILE_DIR/file}"
+        assert (
+            Path(bobfile_temp_dir / file)
+            .read_text()
+            .replace(TEST_PREFIX, "TEST-PREFIX")
+            == (BOBFILE_DIR / file).read_text()
+        )
 
     # And check that the same number of files are created
     new_files = os.listdir(bobfile_temp_dir)
     assert len(old_files) == len(new_files)
+
+
+@pytest.mark.asyncio
+async def test_create_softioc_record_update_send_to_panda(
+    mocked_panda_standard_responses,
+):
+    """Test that updating a record causes the new value to be sent to PandA"""
+    (
+        tmp_path,
+        child_conn,
+        response_handler,
+        command_queue,
+    ) = mocked_panda_standard_responses
+
+    await asyncio.sleep(1)
+    await caput(TEST_PREFIX + ":PCAP1:TRIG_EDGE", "Falling", wait=True, timeout=TIMEOUT)
+    command_queue.put(None)
+    commands_recieved_by_panda = list(iter(command_queue.get, None))
+    assert (
+        command_to_key(Put(field="PCAP1.TRIG_EDGE", value="Falling"))
+        in commands_recieved_by_panda
+    )
+
+
+@pytest.mark.asyncio
+async def test_create_softioc_arm_disarm(
+    mocked_panda_standard_responses,
+):
+    """Test that the Arm and Disarm commands are correctly sent to PandA"""
```

Comment:
This only appears to be testing Arm, not Disarm? Ideal world they'd be two tests, but as you have to set up Arm before you can Disarm I think that's why I originally did both in one.

Reply:
I must have forgotten to add a disarm, doing it now.